### DISCUSSION TITLE
Avoid validating hidden fields when using @hookform/resolvers

### DISCUSSION THREAD

**Discussion by andlbrei** [react-hook-form/react-hook-form#4782] [Q&A]
The default behaviour of RHF is to not apply validation to unregistered fields, even if they are defined as required fields.
I would like to be able to have unregistered fields be exempt from validation when using resolvers, but cannot seem to find a way to do this.
Using resolvers, at least with superstruct and yup, required fields stay required even after unregistration.
This makes sense, but makes validation harder to set up when forms have many conditional fields, sometimes on several levels.

I made two codesanboxes to demonstrate.
The are based on https://codesandbox.io/s/react-hook-form-v6-validationschema-b3dib from the official documentation.

yup: https://codesandbox.io/s/react-hook-form-v6-validationschema-yup-cd5cs?file=/src/index.js
default: https://codesandbox.io/s/react-hook-form-v6-validationschema-basic-6rdv9?file=/src/index.js

Does anyone know of a way to get this behaviour when using resolvers?

Thanks!

**Comment by bluebill1049** [comment]
input hidden is hard to detect, but I am planing to check unmounted input and skip validation on the in-build validation, I think for the schema you may have to use context to support that behavior.

**Reply by andlbrei** [reply]
Thank you for your reply.

I'm not sure what you mean by your plans for "the in-build validation", could you elaborate?

Thanks for the suggestion to use context. I see that not all of the validation libraries has a context concept, but I will investigate yup more.

**Reply by bluebill1049** [reply]
`{false && <input {...regsiter('test', { required: true })}}`

we will be able to skip the validation above since it's not even rendered.

**Reply by andlbrei** [reply]
Yes, but isn't that how it already works when not using resolvers? 
Or is that only in V6? I haven't tried V7 yet, I was waiting for the `shouldUnregister` option to return, which I see is back now.

**Reply by bluebill1049** [reply]
I want to make it work for `shouldUnregister: false` as well and mark inputs as unmounted without removing their value.

**Reply by andlbrei** [reply]
Ah, ok. I understand!
